packages/typespec-azure-resource-manager/lib/private.decorators.tsp

```diff
@@ -82,3 +82,9 @@ extern dec resourceParameterBaseFor(target: ModelProperty, values: unknown[]);
  * Marks an enum as representing the valid `common-types` versions.
  */
 extern dec armCommonTypesVersions(target: Enum);
+
+/**
+ * Provides default name decoration on resource name property with
+ * camelcased and pluralized key and segment name
+ */
+extern dec defaultResourceKeySegmentName(target: ModelProperty, armResource: Model);
```

Comment:
Per discussion, update to take a segmentName parameter and specially process null or empty (or never) values